File: support/pg/src/main/java/com/coffee/gu/PGConfirmResult.java
package com.coffee.gu;

import com.coffee.gu.enums.PaymentMethod;

import java.time.OffsetDateTime;

public record PGConfirmResult(
        String orderId,
        String paymentKey,
        boolean isConfirmed,
        PaymentMethod paymentMethod,
        String approveCode,
        OffsetDateTime approvedAt
) {
    public static PGConfirmResult success(
            String orderId,
            String paymentKey,
            PaymentMethod paymentMethod,
            String approveCode,
            OffsetDateTime approvedAt
    ) {
        return new PGConfirmResult(orderId, paymentKey, true, paymentMethod, approveCode, approvedAt);
    }

    public static PGConfirmResult fail(
            String orderId,
            String paymentKey
    ) {
        return new PGConfirmResult(orderId, paymentKey, false, null, null, null);
    }
}

File: support/pg/src/main/kotlin/com/coffee/gu/PGConfirmResult.kt
package com.coffee.gu

import com.coffee.gu.enums.PaymentMethod
import java.time.OffsetDateTime

data class PGConfirmResult(
    val orderId: String,
    val paymentKey: String,
    val isConfirmed: Boolean,
    val paymentMethod: PaymentMethod? = null,
    val approveCode: String? = null,
    val approvedAt: OffsetDateTime? = null,
) {
    companion object {
        fun success(
            orderId: String,
            paymentKey: String,
            paymentMethod: PaymentMethod?,
            approveCode: String?,
            approvedAt: OffsetDateTime?,
        ): PGConfirmResult = PGConfirmResult(
            orderId = orderId,
            paymentKey = paymentKey,
            isConfirmed = true,
            paymentMethod = paymentMethod,
            approveCode = approveCode,
            approvedAt = approvedAt,
        )

        fun fail(
            orderId: String,
            paymentKey: String,
        ): PGConfirmResult = PGConfirmResult(
            orderId = orderId,
            paymentKey = paymentKey,
            isConfirmed = false,
        )
    }
}
